File: lab3/java/StorageConsumer.java
public class StorageConsumer implements Runnable {
	private final int itemsAmount;
	private final StorageManager manager;
	
	public StorageConsumer (StorageManager manager, int itemsAmount) {
		this.manager = manager;
		this.itemsAmount = itemsAmount;
		
		new Thread(this).start();
	}
	
	@Override
	public void run () {
		for (int i = 0; i < itemsAmount; i++) {
			String item;
			try {
				Thread.sleep(100);
				manager.getTakeItemPermit();
				manager.getAccessPermit();
				
				item = manager.takeItem();
				System.out.println(this + " take : " + item);
				
				manager.dropAccessPermit();
				manager.newAddItemPermit();
			}
			catch (Exception e) {
				e.printStackTrace();
			}
		}
	}
}

File: lab3/cs/StorageConsumer.cs
namespace Laba3_Semaphores_CSharp.classes;

public class StorageConsumer
{
  private int itemsAmount;
  private StorageManager manager;
  private int idx;

  public StorageConsumer(StorageManager manager, int itemsAmount, int idx)
  {
    this.itemsAmount = itemsAmount;
    this.manager = manager;
    this.idx = idx;

    new Thread(Start).Start();
  }

  public void Start()
  {
    for (int i = 0; i < itemsAmount; i++)
    {
      manager.getTakeItemPermit();
      manager.getAccessPermit();

      string item = manager.takeItem();
      Console.WriteLine($"{this.ToString().Split('.')[^1]}-{idx} take : {item}");
                
      manager.dropAccessPermit();
      manager.newAddItemPermit();
    }
  }
}